frontend/packages/console-app/src/components/nodes/NodeTerminal.tsx

```diff
@@ -0,0 +1,156 @@
+
+const getDebugPod = async (name: string, namespace: string, nodeName: string): Promise<PodKind> => {
+  const image = await getDebugImage();
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name,
+      namespace,
+      creationTimestamp: null,
+      annotations: {
+        'debug.openshift.io/source-container': 'container-00',
+        'debug.openshift.io/source-resource': `/v1, Resource=nodes/${nodeName}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [
+        {
+          name: 'host',
+          hostPath: {
+            path: '/',
+            type: 'Directory',
+          },
+        },
+      ],
+      containers: [
+        {
+          name: 'container-00',
+          image,
+          command: ['/bin/sh'],
+          resources: {},
+          volumeMounts: [
+            {
+              name: 'host',
+              mountPath: '/host',
+            },
+          ],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      nodeName,
+      hostNetwork: true,
+      hostPID: true,
+    },
+    status: {
+      phase: 'Unknown',
+    },
+  };
+};
+
+const NodeTerminalError: React.FC<NodeTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const NodeTerminalInner: React.FC<NodeTerminalInnerProps> = ({ obj }) => {
+  const message = (
+    <p>
+      To use host binaries, run <code>chroot /host</code>
+    </p>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return <NodeTerminalError error="The debug pod failed.  Try refreshing the page." />;
+    case 'Running':
+      return <PodExecLoader obj={obj.data} message={message} />;
+    default:
+      return <LoadingBox />;
+  }
+};
+
+const NodeTerminal: React.FC<NodeTerminalProps> = ({ obj: node }) => {
+  const [resources, setResources] = React.useState([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const nodeName = node.metadata.name;
+  React.useEffect(() => {
+    const name = `${node.metadata.name}-debug-${getRandomChars()}`;
+    const namespace = 'default';
+    getDebugPod(name, namespace, nodeName)
+      .then((debugPod: PodKind) => k8sCreate(PodModel, debugPod))
+      .then(() => {
+        setResources([
+          {
+            isList: false,
+            kind: 'Pod',
+            name,
+            namespace,
+            prop: 'obj',
+          },
+        ]);
+      })
+      .catch((e) => setErrorMessage(e.message));
+    const deletePod = () => {
+      // TODO: Only kill pod if it was successfully created.
+      return k8sKillByName(PodModel, name, namespace);
+    };
+    window.addEventListener('beforeunload', () => deletePod());
+    return () => deletePod();
```

Comment:
Actually, you need to remove the event listener as well. So something like

```suggestion
    return () => {
      deletePod().catch((e) => console.warn('Could not delete node terminal pod', e));
      window.removeEventListener('beforeunload', deletePod);
    };
```